components/automate-ui/src/app/components/resource-dropdown/resource-dropdown.component.ts

```diff
@@ -41,7 +41,7 @@ export class ResourceDropdownComponent implements OnInit, OnChanges {
   // filteredResources is merely a container to hold the resources
   // that can be altered
   public filteredResources: ResourceChecked[] = [];
-  public active = false;
+  public dropdownState: 'closed' | 'opening' | 'open' = 'closed';
```

Comment:
Convert this from a binary to a trinary variable, as described in the preamble.

Reply:
Are these standard js terms? Should they be constants? I've seen the string 'open' in a handful of places at this point

Reply:
It's a "string literal type", so that's as good as a constant FWIW. 
 https://mariusschulz.com/blog/string-literal-types-in-typescript has a good intro